Library/Homebrew/formula.rb

```diff
@@ -2081,6 +2081,18 @@ def missing_dependencies(hide: nil)
     []
   end
 
+  # @private
+  sig { returns(T.nilable(String)) }
+  def ruby_source_path
+    path.relative_path_from(tap.path).to_s if path.exist?
+  end
+
+  # @private
+  sig { returns(T.nilable(Checksum)) }
+  def ruby_source_checksum
+    Checksum.new(Digest::SHA256.file(path).hexdigest) if path.exist?
+  end
```

Comment:
```suggestion
  sig { returns(T.nilable(String)) }
  def ruby_source_path
    path.relative_path_from(tap.path).to_s if path&.exist?
  end

  # @private
  sig { returns(T.nilable(Checksum)) }
  def ruby_source_checksum
    Checksum.new(Digest::SHA256.file(path).hexdigest) if path&.exist?
  end
```

It seems like `path` will end up being `nil` if you load it from a source download.

```rb
[13] brew(main)> jrnl = "jrnl".f; nil
=> nil
[14] brew(main)> formula = Homebrew::API::Formula.source_download(jrnl); nil
==> Downloading https://raw.githubusercontent.com/Homebrew/homebrew-core/f6b1d5d773a1c8a7ddffcccb585
Already downloaded: /Users/kevinrobell/Library/Caches/Homebrew/downloads/a849e1f4e3269b75e55c85d56e9840553a665117d667d24dc53f62755455eed2--jrnl.rb
=> nil
[15] brew(main)> formula.to_hash
NoMethodError: undefined method `path' for nil:NilClass
from /usr/local/Homebrew/Library/Homebrew/formula.rb:2087:in `ruby_source_path'
```

This is just an edge case though that users would probably never run into.

Reply:
> This is just an edge case though that users would probably never run into.

IMO it's still worth defending against given that.

Reply:
This also seems like something's gone awry with type-checking if that can happen as Sorbet would normally warn about that.

Reply:
This was actually to do with a nil `tap` - not a nil `path` - and is already resolved in #15291.